src/components/DepositPage.tsx

```diff
@@ -71,15 +81,72 @@ const DepositPage = (props: Props): ReactElement => {
     onConfirmTransaction,
     onToggleDepositWrapped,
   } = props
-
+  const { account, chainId, library } = useActiveWeb3React()
+  const { unstakeMinichef, amountStakedMinichef } = useRewardsHelpers(
+    poolData?.name ?? "",
+  )
   const [currentModal, setCurrentModal] = useState<string | null>(null)
+  const [liquidityGaugeContract, setLiquidityGaugeContract] =
+    useState<LiquidityGaugeV5 | null>(null)
+  const lpTokenContract = useLPTokenContract(poolData?.name ?? "")
   const validDepositAmount = transactionData.to.totalAmount.gt(0)
   const shouldDisplayWrappedOption = isMetaPool(poolData?.name)
   const theme = useTheme()
   const isLgDown = useMediaQuery(theme.breakpoints.down("lg"))
+  const { gauges } = useContext(GaugeContext)
+  const gaugeAddr = gauges?.[poolData?.poolAddress ?? ""]?.address ?? ""
+
+  useEffect(() => {
+    if (!library || !account || !chainId || !poolData || !gaugeAddr) {
+      setLiquidityGaugeContract(null)
+      return
+    }
+    const liquidityGaugeContract = getContract(
+      gaugeAddr,
+      LIQUIDITY_GAUGE_V5_ABI,
+      library,
+      account,
+    ) as LiquidityGaugeV5
+    setLiquidityGaugeContract(liquidityGaugeContract)
+  }, [library, account, chainId, poolData, gauges, gaugeAddr])
 
-  const onMigrateToGaugeClick = () => {
-    // TODO: Hook up call to migrate to gauge.
+  const onMigrateToGaugeClick = async () => {
+    if (
+      !liquidityGaugeContract ||
+      !chainId ||
+      !account ||
+      !poolData ||
+      !lpTokenContract
+    )
+      return
+    try {
+      await unstakeMinichef(amountStakedMinichef)
+      await checkAndApproveTokenForTrade(
+        lpTokenContract,
+        liquidityGaugeContract.address,
+        account,
+        await lpTokenContract.balanceOf(account),
+        true,
+        Zero, // @dev: gas not being used
+        {
+          onTransactionError: () => {
+            throw new Error("Your transaction could not be completed")
+          },
+        },
+        chainId,
+      )
+      const txn = await liquidityGaugeContract["deposit(uint256,address,bool)"](
+        await lpTokenContract.balanceOf(account),
```

Comment:
will this pretty much always be the same value as the last `await lpTokenContract.balanceOf(account)` in this block? can this be saved as a var to avoid calling twice?

Reply:
these are diff. one is balance in the liq gauge. the deposit call is pulling balance from lpTokenContract.

Reply:
One is providing a current balance of lpTokens in the user's wallet. The other is getting the user's balance of lpToken within the gauge.